src/components/pages/gallery/Upload.tsx

```diff
@@ -86,30 +92,50 @@ export default function Upload(props: Props) {
         );
     }, []);
 
+    useEffect(() => {
+        isElectron() &&
+            ImportService.getIfToUploadFilesExists().then((exists) => {
+                if (exists) {
+                    uploadFailedFiles();
```

Comment:
we can just call `setElectronFiles(getPendingUploads())` Where `getPendingUploads` would just return the electronFile for files in the pendingUpload list?

Reply:
I think we need the `collectionName` too.

Reply:
ya ya what I meant was we can avoid the two-step process of first getting the filepath and then calling  the getElectronFile Function to get the  ElectronFile instance,  instead we can directly send the `ElectronFile` instance from the electron layer na.

we can send this as function response
```
{
collectionName: string,
files:ElectronFile[]
}
```